lib/kubernetes-deploy/concurrency.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  module Concurrency
+    MAX_THREADS = 8
+
+    def self.split_across_threads(all_work, &block)
+      return if all_work.empty?
+      raise ArgumentError, "Block of work is required" unless block_given?
+
+      slice_size = ((all_work.length + MAX_THREADS - 1) / MAX_THREADS)
+      threads = []
+      all_work.each_slice(slice_size) do |work_group|
```

Comment:
(it should be set in the parent thread, as per example in http://ruby-doc.org/core-2.4.1/Thread.html#method-c-abort_on_exception-3D )